IssueTracker/IssueTracker/Resource/Repository/LoginRepository.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  LoginRepository.swift
+//  IssueTracker
+//
+//  Created by 최예주 on 2022/06/17.
+//
+
+import Foundation
+import UIKit
+
+class LoginRepository {
+
+    func requestGithubAuthorize() {
+        let clientID = Environment.clientId
+        let scope = Environment.scope
+        let urlString = "https://github.com/login/oauth/authorize"
+        guard var components = URLComponents(string: urlString) else { return }
+        components.queryItems = [
+            URLQueryItem(name: "client_id", value: clientID),
+            URLQueryItem(name: "scope", value: scope)
+        ]
+        guard let url = URL(string: "\(components.url!)"), UIApplication.shared.canOpenURL(url) else { return }
+
+        UIApplication.shared.open(url)
+    }
+
+    func getGithubAccessCode(_ target: BaseTarget, _ completion: @escaping (String) -> Void) {
+
+        guard let request = RequestRepository.makeURLRequest(with: target) else { return }
+        RequestRepository.sendRequest(with: request) { data in
+            guard let jsonData = try! JSONSerialization.jsonObject(with: data) as? [String: Any] else { return }
```

Comment:
1. guard 문에서 실패 케이스에 대해서 로직을 따로 처리하지 않는다면 한줄로 작성 가능합니다.

```suggestion
            guard let jsonData = try? JSONSerialization.jsonObject(with: data) as? [String: Any], 
                       let let access_token = jsonData["access_token"] as? String else { 
                       return 
            }
```

2. parsing 실패하거나 access_token 값을 가져오지 못할 때 예외 처리는 따로 하지 않아도 괜찮을까요?

Reply:
> parsing 실패하거나 access_token 값을 가져오지 못할 때 예외 처리는 따로 하지 않아도 괜찮을까요?
다음 PR 때, 로그인 에러 타입을 만들어서 따로 에러 처리 하도록 구현할 예정입니다!